backend/internal/service/user.go

```diff
@@ -53,70 +51,56 @@ func (s UserService) GetAllUsers() ([]response.User, error) {
 }
 
 func (s UserService) GetUserById(id string) (response.User, error) {
-	idx, exists := s.repo.GetUserIdxById(id)
-	if !exists {
+	user, err := s.repo.GetUser(id)
+	if err != nil {
 		return response.User{}, errors.New("No such user")
 	}
 
-	user, err := s.repo.GetUserByIdx(idx)
-	return user.ToResponse(), err
+	return user.ToResponse(), nil
 }
 
 func (s UserService) GetUserByEmail(email string) (response.User, error) {
-	idx, exists := s.repo.GetUserIdxByEmail(email)
-	if !exists {
+	user, err := s.repo.GetUserByEmail(email)
+	if err != nil {
 		return response.User{}, errors.New("No such user")
 	}
 
-	user, err := s.repo.GetUserByIdx(idx)
-	return user.ToResponse(), err
+	return user.ToResponse(), nil
 }
 
-func (s UserService) CreateUser(user request.User) (response.User, error) {
-	_, exists := s.repo.GetUserIdxByEmail(user.Email)
-	if exists {
-		return response.User{}, errors.New("User already exists")
-	}
-
-	hash, err := bcrypt.GenerateFromPassword([]byte(user.Password), 10)
+func (s UserService) CreateUser(data request.User) (response.User, error) {
+	hash, err := bcrypt.GenerateFromPassword([]byte(data.Password), 10)
```

Comment:
it's not secure enough to just calculate hash, consider adding salt
https://www.vaadata.com/blog/how-to-securely-store-passwords-in-database/
https://www.geeksforgeeks.org/store-password-database/